File: Data Structures & Algorithms/sliding-window-maximum/submission-3.java
class Solution {
    static int[] maxSlidingWindow(int[] nums, int k) {
      int[] maxElems = new int[nums.length - k + 1];

      Deque<Integer> deque = new ArrayDeque<>();

      for (int r = 0, l = r - k + 1; r < nums.length; r++, l++) {
         while (!deque.isEmpty() && deque.peek() < l) {
            deque.pollFirst();
         }

         while (!deque.isEmpty() && nums[deque.peekLast()] <= nums[r]) {
            deque.pollLast();
         }

         deque.add(r);

         if (l >= 0) {
            maxElems[l] = nums[deque.peek()];
         }
      }

      return maxElems;
   }
}

File: Data Structures & Algorithms/two-integer-sum/submission-3.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        i = 0
        j = len(nums) - 1

        while i < j:
            sum = nums[i] + nums[j]
            if sum < target:
                i = i + 1
            elif sum > target:
                j = j - 1
            # sum == target
            else:
                return [i, j]